pkg/broker/filter/filter_handler.go

```diff
@@ -207,6 +207,26 @@ func (h *Handler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
 		err = h.tokenVerifier.VerifyJWTFromRequest(ctx, request, &audience, writer)
 		if err != nil {
 			h.logger.Warn("Error when validating the JWT token in the request", zap.Error(err))
+
+			// If the error is due to the absence of a JWT token, report it as an unauthenticated request.
+			// The error message is "no JWT token found in request"
+			if errors.Is(err, auth.ErrNoJWTTokenFound) {
+				_ = h.reporter.ReportUnauthenticatedRequest(&ReportArgs{
```

Comment:
How do you think the way here to distinguish these 2 different scenario? Are there any better approach you would recommend? @creydr

Reply:
I think this can work. You could also think about passing the reporter to the tokenVerifier, so this reports the metrics on each request and not only do this in the caller :thinking: